test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,901 @@
-//             await expect(
-//                 execute(
-//                     "VaultRegistry",
-//                     { from: stranger2, autoMine: true },
-//                     "transferFrom",
-//                     stranger,
-//                     randomAddress(),
-//                     nft
-//                 )
-//             ).to.be.revertedWith(Exceptions.LOCK);
-//         });
-
-//         describe("when called not by owner", () => {
-//             it("reverts", async () => {
-//                 const { execute, read, get } = deployments;
-//                 const { stranger, stranger2, deployer } =
-//                     await getNamedAccounts();
-//                 const vault = randomAddress();
-//                 const vaultGovernance = await get("ERC20VaultGovernance");
-//                 await withSigner(vaultGovernance.address, async (s) => {
-//                     const vaultRegistry = await ethers.getContract(
-//                         "VaultRegistry"
-//                     );
-//                     await vaultRegistry
-//                         .connect(s)
-//                         .registerVault(vault, stranger);
-//                 });
-//                 const nft = await read("VaultRegistry", "vaultsCount");
-//                 await execute(
-//                     "VaultRegistry",
-//                     { from: stranger, autoMine: true },
-//                     "approve",
-//                     stranger2,
-//                     nft
-//                 );
-
-//                 for (const actor of [stranger2, deployer]) {
-//                     await expect(
-//                         execute(
-//                             "VaultRegistry",
-//                             { from: actor, autoMine: true },
-//                             "lockNft",
-//                             nft
-//                         )
-//                     ).to.be.revertedWith(Exceptions.FORBIDDEN);
-//                 }
-//             });
-//         });
-//     });
-// });
+import { expect } from "chai";
+import { ethers, deployments } from "hardhat";
+import {
+    addSigner,
+    now,
+    randomAddress,
+    sleep,
+    sleepTo,
+    withSigner,
+} from "./library/Helpers";
+import { setupDefaultContext, TestContext } from "./library/setup";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signers";
+import { ProtocolGovernance, VaultRegistry } from "./types";
+import { Contract } from "ethers";
+import { Address } from "hardhat-deploy/dist/types";
+import { REGISTER_VAULT } from "./library/PermissionIdsLibrary";
+import { address, pit, RUNS } from "./library/property";
+import { Arbitrary, integer } from "fast-check";
+import Exceptions from "./library/Exceptions";
+import { boolean } from "hardhat/internal/core/params/argumentTypes";
+
+type CustomContext = {
+    strategySigner: SignerWithAddress;
+    ownerSigner: SignerWithAddress;
+    newProtocolGovernance: Contract;
```

Comment:
It is inconvenient to make newProtocolGovernance local, because it is used in various descriptions. However it is possible to make global context smaller